developer/developer-guide.md

```diff
@@ -1,3 +1,872 @@
+has Alfred API installed.
+
+# Java Api
+
+The alfred-api java api the core to exposing alfresco functionality and normalizing operations across version.
+Any extensions written while depending on alfred api can be easily ported to a new alfresco version.
+
+When the api is installed, all of its service are available as beans and can be wired into your own classes.
+
+# Rest Api
+
+For a full overview of the Rest api, please refer to [the swagger specification](https://demo.xenit.eu/alfresco/s/apix/v1/docs/ui.html).
+
+## Search Requests
+
+### query
+Object containing subcomponents that build the requested query.
+All queries are translated to Alfresco Full Text Search ([AFTS]("https://community.alfresco.com/docs/DOC-5729-full-text-search-query-syntax")) by alfred-api when executed
+
+#### Syntax
+
+The query parameter takes a tree structure of searchnodes as its argument.
+A searchnode is either an operator or a search term.
+
+##### Operators
+
+Operators currently include only the standard AND, OR & NOT logical operations.
+An operator is structured in the json payload as a named list.
+
+* And
+
+Translating `cm:name:"Budget 2020.xls" AND example:customProperty:"#AA3BF5"`
+```json
+{
+  "and": [
+    {
+      "property" : {
+        "name": "cm:name",
+        "value": "Budget 2020.xls"
+      }
+    },
+    {
+      "property" : {
+        "name" : "example:customProperty",
+        "value" : "#AA3BF5"
+      }
+    }
+  ]
+}
+```
+
+* Or
+
+Translating `example:customProperty:"#FFF233" OR example:customProperty:"#AA3BF5"`
+```json
+{
+  "or": [
+    {
+      "property" : {
+          "name" : "example:customProperty",
+          "value" : "#FFF233"
+        }
+    },
+    {
+      "property" : {
+        "name" : "example:customProperty",
+        "value" : "#AA3BF5"
+      }
+    }
+  ]
+}
+```
+
+* Not
+
+Translating `NOT cm:name:"Budget 2020.xls"`
+```json
+{
+  "not": [
+    {
+      "property" : {
+          "name": "cm:name",
+          "value": "Budget 2020.xls"
+        }
+    }
+  ]
+}
+```
+
+* Nesting
+
+Operators can be nested to form complex queries:
+```json
+{
+  "and" : [
+    {
+      "not": [
+         {
+            "property" : {
+                "name" : "example:customProperty",
+                "value" : "#FFF233"
+             }
+          }
+      ]
+    },
+    {
+      "or": [
+          {
+            "property" : {
+                "name" : "example:customProperty",
+                "value" : "#FFF233"
+              }
+          },
+          {
+            "property" : {
+              "name" : "example:customProperty",
+              "value" : "#AA3BF5"
+            }
+          }
+      ]
+    }
+  ]
+}
+```
+##### Search terms
+
+Search terms can be split into two groups: special terms and generic terms.
+Special terms are used to search for specific concepts, generic terms are used to search for property values at large.
+
+###### Special terms
```

Comment:
These headers do not render in the PDF. Either `######` does not work, in case find an alternative (bold text or a different organization) or maybe it's a Markdown problem.